_stbt/core.py

```diff
@@ -1948,28 +1946,25 @@ def pull_frame(self, timeout_secs=10):
 
     def frames(self, timeout_secs=None):
         import time
-        self.start_timestamp = None
+        if timeout_secs is not None:
+            end_time = time.time() + timeout_secs
```

Comment:
This is also a change in behaviour in that the timeout is based on wall clock time instead of the time of the first frame.

It will break video-capture devices that don't use wall-clock time. I think that's OK -- as far as I know that's only the HD-PVR which is obsolete.

It will change test-script behaviour for video-capture devices with significant latency --  but in a safe way, i.e. it will result in more frames being processed, not less. So if you ask for timeout_secs=10, you'll actually get 10 seconds' worth since your `press` or whatever.